src/pages/Detail/DetailTop.js

```diff
@@ -1,54 +1,121 @@
 import React, { Component } from 'react';
 import { Link } from 'react-router-dom';
 
+import AmountOption from './AmountOption';
+
+// select -> state
+// 함수를 만들어서 자식한테 전달
+// 자식에서 함수기 실행되면은 select된 값을 변경
+
 export class DetailTop extends Component {
+  //state초기화
+  constructor(props) {
+    super(props);
+    this.state = {
+      numValue: 1,
+      isOn: false,
+      amountOption: [],
+      selectedOption: this.props.productOption[0],
+      productOption: '',
+      option: '',
+    };
+    this.onClickToggleHandler = this.onClickToggleHandler.bind(this);
+  }
+  //수량 증가 감소
+  onClickHandler = () => {
+    this.setState({ numValue: this.state.numValue + 1 });
+  };
+  onClickhandlerRemove = () => {
+    //const { numValue } = this.state.numValue;
+    if (this.state.numValue > 1) {
+      this.setState({ numValue: this.state.numValue - 1 });
+    }
+  };
+  //종류 추가
+  onClickToggleHandler = () => {
+    //const { isOn } = this.state.isOn;
+    //console.log(isOn);
+    this.setState({
+      isOn: !this.state.isOn,
+    });
+  };
+
+  selectOption = option => {
+    this.setState({
+      selectedOption: option,
+    });
+    this.onClickToggleHandler();
+  };
+
   render() {
+    const productInfo = this.props;
```

Comment:
- 다음과 같이 구조분해 할당 해주셔서 사용해주세요!
```suggestion
    const { productImg, productName, productPrice, 나머지 것들도 } = this.props;
```

Reply:
해당 부분들 this.props로 수정하였습니다,뭔가 부모 요소에서 보내주는것 못받길래....확인한번 부탁드리겠습니다.